### PR TITLE
test: Use JDK Files API in PodIT tests

### PR DESCRIPTION
**Pull request by jorsol**
## Description

Change from Guava/Apache API to pure JDK Files API in the PodIT test

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java [hunk 1]
```diff
@@ -253,55 +255,48 @@ public void uploadFile() throws IOException {
...
 
     Stream.of(files).forEach(fileName -> {
-      try (
-        final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
-          .withName(pod1.getMetadata().getName()).file("/tmp/uploadDir/"+fileName).read();
-        final ByteArrayOutputStream resultOs = new ByteArrayOutputStream()
-      ) {
-        IOUtils.copy(checkIs, resultOs);
-        assertEquals(resultOs.toString(StandardCharsets.UTF_8.name()),"I'm uploaded");
+      try (final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java:295]: Could we move the Reader instantiation to the resources block so it gets closed automatically.

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java [hunk 2]
```diff
@@ -253,55 +255,48 @@ public void uploadFile() throws IOException {
...
+          .withName(pod1.getMetadata().getName()).file("/tmp/uploadDir/" + fileName).read()) {
+        String result = new BufferedReader(new InputStreamReader(checkIs, StandardCharsets.UTF_8))
+          .lines().collect(Collectors.joining(System.lineSeparator()));
+        assertEquals("I'm uploaded", result);
       } catch(IOException ex) {
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java:300]: Could we remove the catch blocks?

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java [hunk 3]
```diff
@@ -315,13 +310,14 @@ public void copyFile() throws IOException {
...
-    client.pods().inNamespace(session.getNamespace()).withName(pod1.getMetadata().getName()).file("/msg").copy(tmpDir.toPath());
-    File msg = tmpDir.toPath().resolve("msg").toFile();
-    assertTrue(msg.exists());
-    try (InputStream is = new FileInputStream(msg))  {
-      String result = new BufferedReader(new InputStreamReader(is)).lines().collect(Collectors.joining("\n"));
+    client.pods().inNamespace(session.getNamespace()).withName(pod1.getMetadata().getName()).file("/msg").copy(tmpDir);
+    Path msg = tmpDir.resolve("msg");
+    assertTrue(Files.exists(msg));
+    try (InputStream is = Files.newInputStream(msg)) {
+      String result = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java:319]: See https://github.com/fabric8io/kubernetes-client/pull/2879/files#r589333608

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java [hunk 4]
```diff
@@ -253,55 +255,48 @@ public void uploadFile() throws IOException {
...
-        .withName(pod1.getMetadata().getName()).file("/tmp/toBeUploaded").read();
-      final ByteArrayOutputStream resultOs = new ByteArrayOutputStream()
-    ) {
-      IOUtils.copy(checkIs, resultOs);
-      assertEquals(resultOs.toString(StandardCharsets.UTF_8.name()),"I'm uploaded");
+      .file("/tmp/toBeUploaded").upload(tmpFile);
+
+    try (final InputStream checkIs = client.pods().inNamespace(session.getNamespace())
+        .withName(pod1.getMetadata().getName()).file("/tmp/toBeUploaded").read()) {
+      String result = new BufferedReader(new InputStreamReader(checkIs, StandardCharsets.UTF_8))
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java:266]: https://github.com/fabric8io/kubernetes-client/pull/2879/files#r589333608

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**